src/commands/sns_canisters_summary.rs

```diff
@@ -0,0 +1,39 @@
+use crate::lib::signing::{sign_ingress_with_request_status_query, IngressWithRequestId};
+use crate::lib::{AnyhowResult, TargetCanister};
+use crate::SnsCanisterIds;
+use candid::Encode;
+use clap::Parser;
+use ic_base_types::{CanisterId, PrincipalId};
+use std::str::FromStr;
+
+#[derive(Parser, Debug)]
+#[clap(about, version, author)]
+pub struct SnsCanistersSummaryOps {
+    /// Canister id of the dapp.
+    #[clap(long)]
+    pub canister_id: Option<String>,
```

Comment:
Since we're now requiring that the canister ids be made available in the canister_ids.json file, I don't think this is needed anymore ad can be removed.

Reply:
I initially changed to optional so that the dapp canister id can be specified in the command if it is not in the json file. However if we require it I will remove it from here.